app/assets/javascripts/courses.js

```diff
@@ -0,0 +1,72 @@
+var shown = false;
+var current_student = {};
+// called when the "draw flashcard" button is pressed
+function drawFlashcard(source) {
+  // send an AJAX GET request to the flashcard controller
+  $.get(flashcard_url, function(data, status){
+    if (status == "success") {
+      // send the returned student object to the populate function
+      populateFlashcard(data, source);
+    }

```

Comment:
If the status is not success, theoretically something went wrong. Should you manually throw an error, or does javascript/jquery do that automatically?

Reply:
Many things could go wrong to make the status not a success, including an internet connection problem or anything else on the client side. Since we can't be sure an error is a server-side problem with our stuff, it seems to me that it's better to fail silently.